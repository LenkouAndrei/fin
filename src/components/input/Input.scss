@import "../../styles/variables.scss";

.ipt {
    font-size: 1.8rem;
    color: $graphit;
    line-height: 3rem;
    border-radius: 10px;
    border: none;
    width: 400px;
    box-sizing: border-box;
    padding-left: 15px;
}