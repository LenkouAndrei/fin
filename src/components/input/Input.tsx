import React from 'react';
import './Input.scss';

interface IInputProps {
  children?: string | number;
  type?: string;
  placeholder?: string;
  onChange?(args?: unknown): void
}

export const Input: React.FC<IInputProps> = ({
  children,
  onChange = () => {},
  type = 'text',
  placeholder = 'type info',
}: IInputProps) => {
  return <input onChange={onChange} className="ipt" type={type} placeholder={placeholder} value={children} />;
};
