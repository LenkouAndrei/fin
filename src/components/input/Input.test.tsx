import { render, screen, fireEvent } from '@testing-library/react';
import { Input } from './Input';

describe('<Input />', () => {
  it('should render Input', () => {
    const { asFragment } = render(<Input>Test</Input>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call passed callback on change', () => {
    const mockCallback = jest.fn();
    render(<Input onChange={mockCallback}>Hello</Input>);
    const inputEl = screen.getByDisplayValue('Hello');
    fireEvent.change(inputEl, { target: { value: 'a' } });
    expect(mockCallback).toHaveBeenCalled();
  });

  it('should apply passed type to input', () => {
    const typeAttr = 'number';
    render(<Input type={typeAttr}>2</Input>);
    const inputEl = screen.getByDisplayValue('2');
    const inputAttr = inputEl.getAttribute('type');
    expect(inputAttr).toBe(typeAttr);
  });

  it('should apply passed placeholder to input', () => {
    const placeholder = 'number';
    render(<Input placeholder={placeholder}>Hello</Input>);
    const inputEl = screen.getByPlaceholderText(placeholder);
    expect(inputEl).toBeTruthy();
  });
});
