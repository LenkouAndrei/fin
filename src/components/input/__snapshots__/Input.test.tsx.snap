// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render Input 1`] = `
<DocumentFragment>
  <input
    class="ipt"
    placeholder="type info"
    type="text"
    value="Test"
  />
</DocumentFragment>
`;
